lib/sampler.py

```diff
@@ -142,11 +155,19 @@ def _sample(self):
             except ProcessLookupError:
                 self.unregister_process(pid)
             else:
-                self._sample_process(pid)
-
-    def _sample_process(self, pid):
-        # Your sampling code here.
-        pass
+                self._sample_process(pid, tasks_rss)
+
+        # Save current overall RSS value if it is bigger than saved.
+        for task_id in tasks_rss:
+            if task_id not in self._rss_summary or \
+                    self._rss_summary[task_id] < tasks_rss[task_id]:
+                self._rss_summary[task_id] = tasks_rss[task_id]
+
+    def _sample_process(self, pid, tasks_rss):
+        task_id = self._processes[pid]['task_id']
+        # Count overall RSS per task.
+        tasks_rss[task_id] = get_proc_stat_rss(pid) + \
+            (tasks_rss[task_id] if task_id in tasks_rss else 0)
```

Comment:
And here we can write it shorter too: `<...> + tasks_rss.get(task_id, 0)`.

Reply:
Sure, changed.